fire/value_types.py

```diff
@@ -37,7 +37,8 @@ def IsCommand(component):
 
 
 def IsValue(component):
-  return isinstance(component, VALUE_TYPES)
+  return isinstance(component, VALUE_TYPES) or (
+      hasattr(component, '__str__') and inspect.ismethod(component.__str__))
```

Comment:
I think we're going to do this a little differently, using the same logic as in core.py:_PrintResult to determine what counts has having a custom __str__ method.

No need to make any changes though; we'll accept your PR and then add a follow-on commit with the change.

Reply:
Sounds good - thanks!